src/lucky/request.cr

```diff
@@ -0,0 +1,72 @@
+require "http/client"
+
+class Lucky::Request
+  getter! url
+  getter! headers
+  getter! query_params
+  getter! response
```

Comment:
I’ve been thinking about these getters and I think we can do something so they can be removed. The advantage is that we don’t need the ! Which means we have a bit better type safety. 

What I think could be done is to set up a TestServer like we did in LuckyFlow: 
https://github.com/luckyframework/lucky_flow/blob/3878da1bead09df01f65dc55f755c09b28ba7598/spec/spec_helper.cr#L10

https://github.com/luckyframework/lucky_flow/blob/master/spec/support/test_server.cr

It would probably work a bit differently in that the TestServer would record the HTTP method, headers and path so it can be inspected. 

Then in the specs you can make a real request to the test server instead of testing that the instance variables are set in Lucky:mRequest. In fact, I think you could remove the instance variables since they don’t need to be accessed outside the class anymore. 

Lmk if this makes sense. It’s a bit tricky to implement and I may nit have explained it well. I’m happy to go more in depth if that would help.

Reply:
Oh and here’s an example of using the test server in LuckyFlow to stub requests: 
https://github.com/luckyframework/lucky_flow/blob/master/spec/lucky_flow_spec.cr

Reply:
From what I understand, we're wrapping the server creation in a new class, then capturing the requests we make to it so we can inspect them later.

So the class would look something like this:

```crystal
class TestServer
  delegate listen, close, to: @server

  def initialize(port : Int32)
    @server = App.new
  end

  def self.request
    # return last request
  end

  def self.response
    # return last response
  end
end
```
And we would use it like `TestServer.response.headers["Authorization"].should contain "token"`. 

Is this what you had in mind?

Reply:
Yeah I think that should work!